tests/KAT_MiyaguchiPreneel.hs

```diff
@@ -0,0 +1,53 @@
+
+module KAT_MiyaguchiPreneel (tests) where
+
+import           Crypto.Cipher.AES (AES128)
+import           Crypto.ConstructHash.MiyaguchiPreneel
+
+import           Imports
+
+import           Data.Char (digitToInt)
+import qualified Data.ByteString as BS
+import qualified Data.ByteArray as B
+
+
+runMP128 :: ByteString -> ByteString
+runMP128 s = B.convert (mp s :: MiyaguchiPreneel AES128)
+
+hxs :: String -> ByteString
+hxs = BS.pack . rec' where
```

Comment:
there's a Data.ByteArray.Encoding module that gives you Base16 decoding/encoding for free.